tools/dynamodbConnector.py

```diff
@@ -12,63 +12,61 @@ class DynamoDBConnector:
     def __init__(self):
         """Initialise the connection, and create a logger instance."""
         self.logger = logging.getLogger(__name__)
-        self.dynamodb = boto3.resource(
+        self.dynamodb = boto3.client(
             'dynamodb',
-            region_name='eu-west-2',
-            endpoint_url='http://localhost:8765'
         )
         try:
-            if 'scraper_articles' not in self.dynamodb.tables.all():
-                self.logger.error('Article table does not exists.')
+            tables = self.dynamodb.list_tables().get('TableNames')
         except ClientError as e:
             self.logger.error(
                 'Error when initialising the connection [%s]', e
             )
+        else:
+            if 'scraper_articles' not in tables:
```

Comment:
not sure of best practice or efficiency on that but it might be quicker to try to access the table and catch the exception in case tables does not exist.

Reply:
and btw if the reason is that you want to catch multiple exceptions, then you can break down the code in two lines in try and have multiple excepts catching different exceptions and logging different errors

Reply:
☝️ 

Best practice in Python is that it’s _Easier to Ask Permission than Forgiveness_ (try it, catch the exception if it fails) rather than _Look Before You Leap_ (check to see if it’s correct).

The former also protects you from a race condition – if you check the table exists, it does, and then it gets deleted before you do the lookup, you’ll hit the same error anyway.

Reply:
I’d have to double-check, but I think the DynamoDB issues a different error for “table not found” and other failure modes.